themes-default/slim/static/js/router.js

```diff
@@ -0,0 +1,47 @@
+const VueRouter = window.vueRouter;
+const { routes, httpVueLoader } = window;
+
+if (!window.router) {
+    const configComponent = httpVueLoader('js/templates/config.vue');
+    const notFoundComponent = httpVueLoader('js/templates/http/404.vue');
+
+    const router = new VueRouter({
+        base: document.body.getAttribute('api-root').replace('api/v2/', ''),
+        mode: 'history',
+        routes: Object.assign(routes || [], [{
+            path: '/config',
+            name: 'config',
+            meta: {
+                title: 'Help & Info',
+                header: 'Medusa Configuration'
+            },
+            component: configComponent
+        }, {
+            path: '/not-found',
+            name: 'not-found',
+            meta: {
+                title: '404',
+                header: '404 - page not found'
+            },
+            component: notFoundComponent
+        }, {
+            path: '*',
+            redirect: '/not-found'
```

Comment:
Not getting this issue on dev with and without webroot so I'm not sure what's causing it on your end.Do you maybe still have the service worker loaded?

Reply:
I don't use a webroot. ServiceWorker is loaded but I don't it has anything to do with it.

Try navigating from say, `/home` to `/schedule` by clicking the Schedule nav link. You should get this issue.
Don't know why it only seems to happen on `/schedule` route.

Reply:
That 100% has something todo with it. Remove the service worker.

Reply:
Oh, I just figured it out.
https://github.com/pymedusa/Medusa/blob/9eaf64841e85608af4ea1cff3a516237a9ef7bed/themes/light/templates/schedule.mako#L15
We need to remove this line.

Reply:
Oh.. no we just need to add the route for it. I'm moving the headers and titles over to the `router.js`. Just don't add a component property and it works fine.

For example.

```js
routes: Object.assign(routes || [], [{
    path: '/config',
    name: 'config',
    meta: {
        title: 'Help & Info',
        header: 'Medusa Configuration'
    },
    component: configComponent
}, {
    path: '/home',
    name: 'home',
    meta: {
        title: 'Home',
        header: 'Show list'
    }
}, {
    path: '/history',
    name: 'history',
    meta: {
        title: 'History',
        header: 'History'
    }
}, {
    path: '/not-found',
    name: 'not-found',
    meta: {
        title: '404',
        header: '404 - page not found'
    },
    component: notFoundComponent
}, {
    path: '*',
    redirect: '/not-found'
}])
```

Reply:
We then use this in the pages that aren't `.vue` since they don't use the global Vue.

```
header() {
    return this.$route.meta.header;
},
```

Reply:
This is the only page that has the `router` as a property now.
I still think removing it is the correct option since it'll be a while until schedule is moved to the router.